### PR TITLE
feat: full compatibility with neovim v0.6

### PR DESCRIPTION
**Pull request by kylo252**
# Description

- Add some missing handlers configuration for full compatbility with Neovim v0.6
- Remove code that specific to v0.5.0 considering that most plugins don't support it anymore
- Add a check in the installer to verify that v0.5.1 or higher is available
- bump treesitter version with `vim.cmd` parsing support

## How Has This Been Tested?

Using `gl` now uses the new `vim.diagnostic.open_float()` call.

### CODE REVIEW

**Comment by abzcoding** [comment]
not sure why, but i'm getting this error, even after `:PackerSync` and `:LvimCacheReset`
Screen Shot 2021-12-05 at 4 32 33 PM

**Comment by kylo252** [comment]
> not sure why, but i'm getting this error, even after `:PackerSync` and `:LvimCacheReset` Screen Shot 2021-12-05 at 4 32 33 PM

Yeah, it's normal in my experience, just need to do `:PackerCompile` at least once or twice, and make sure to do :TSUpdate` which I've already added in post-update hook.

**Comment by abzcoding** [comment]
ok everything is falling apart now 😢 , let me see what is going on and get back to you
Screen Shot 2021-12-05 at 4 39 14 PM


just to be sure, i'm on mac and this is my neovim version
Screen Shot 2021-12-05 at 4 42 11 PM

**Comment by abzcoding** [comment]
@kylo252 yeah the issue was from my side, its fixed now
let me make sure there are not any other issues as well

File: lua/lvim/lsp/config.lua
Comment (abzcoding) [inline lua/lvim/lsp/config.lua:18]: should we add
```lua
      border = "rounded",
      focusable = false,
```
here as well or you prefer the current one?

Comment (kylo252) [reply lua/lvim/lsp/config.lua:18]: ~~do we have a border setting somewhere that we can re-use?~~
I'll switch it to use `lvim.lsp.popup_border`

**Review by abzcoding — APPROVED** [review]
tyvm <3

**Comment by kylo252** [comment]
I'll leave it until tomorrow or until I get a second review. Just in case we missed something.

**Pull request merged**
